[master] Add heap dump docs for CC Router and Use placeholder for Choreo Connect version (#6110)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md

## Taking a heap dump of Enforcer

Enforcer is implemented in Java and it is configured with required parameters to dump the heap information when out of memory errors happen. The dump can be collected from `/home/wso2/logs/heap-dump.hprof`

## Taking a heap dump of Router

> Router uses the Envoy Proxy as the core component that does the traffic routing. The default Router which is used in you production environment is not supporting heap profiling and we have built a separate Router Docker image which enables heap profiling. Follow the steps below to take a heap profile of the Router.

Reviewer: ```suggestion
The Router uses the Envoy Proxy as the core component that does the traffic routing. The default Router that is used in your production environment does not support heap profiling and we have built a separate Router Docker image that enables heap profiling. Follow the steps below to take a heap profile of the Router.
```

### Prerequisites

1.  Install [pprof](https://github.com/google/pprof#readme) and [Graphviz](http://www.graphviz.org/) in your local machine. **pprof** is used to analyze the heap profile and **Graphviz** enables to analyze them with graphs.

2.  Build ***Router Debug Image***, if you are taking heap profile **on top of an updated Router image**.

    !!! note

>         For the **GA version** of the Router docker image you can use `wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}}` as the ***Router Debug Image***.

Reviewer: ```suggestion
        For the **GA version** of the Router Docker image, you can use `wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}}` as the **Router Debug Image**.
```

>     1.  Create a Dockerfile as follows. Here we create a custom docker image by replacing the envoy binary in default docker image with heap profile enabled envoy binary.

Reviewer: ```suggestion
    1.  Create a Dockerfile as follows. Here we create a custom Docker image by replacing the envoy binary in default Docker image with heap profile enabled envoy binary.
```

        <!-- Following docker image "wso2am/choreo-connect-router" contains envoy version in its tag, it should be updated per router envoy version update -->

        ```Dockerfile tab="Format"
        FROM <ROUTER_DOCKER_IMAGE_WITH_WSO2_UPDATES>
        COPY --from=wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}} /usr/local/bin/envoy /usr/local/bin/envoy
        ```

        ```Dockerfile tab="Sample Dockerfile"
        FROM wso2/choreo-connect-router:{{choreo_connect.version}}.1 # for update level 1
        COPY --from=wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}} /usr/local/bin/envoy /usr/local/bin/envoy
        ```

>     2. Build an docker image and push it to a Docker registry that your K8s cluster can pull it.

Reviewer: ```suggestion
    2. Build a Docker image and push it to a Docker registry that your Kubernetes cluster can pull it.
```
